core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -358,7 +358,6 @@ public PlanOptimizers(
                 new PruneWindowColumns());
 
         Set<Rule<?>> projectionPushdownRules = ImmutableSet.of(
```

Comment:
what if we keep the code here unchanged and just add rules
```
                new PushProjectionThroughUnion(),
                new PushProjectionThroughExchange(),
```
explicitly at
`io/trino/sql/planner/PlanOptimizers.java:438`
Then I think we don't have to change anything else in `PlanOptimizers`

Reply:
It solves the problem for table scan redirection but I'm not sure whether it breaks anything for dereference pushdown